src/MSBuildLocator/MSBuildLocator.cs

```diff
@@ -209,6 +209,25 @@ public static void RegisterMSBuildPath(string[] msbuildSearchPaths)
             // AssemblyResolve event can fire multiple times for the same assembly, so keep track of what's already been loaded.
             var loadedAssemblies = new Dictionary<string, Assembly>();
 
+            // MSBuild can be loaded from the x86 or x64 folder. Before 17.0, it looked next to the executing assembly in some cases and constructed a path that assumed x86 in others.
+            // This overrides the latter assumption to let it find the right MSBuild.
+            foreach (string path in msbuildSearchPaths)
+            {
+                string msbuildExe = Path.Combine(path, "MSBuild.exe");
+                if (File.Exists(msbuildExe))
+                {
+                    if (FileVersionInfo.GetVersionInfo(msbuildExe).FileMajorPart < 17)
```

Comment:
Should we do an exact check on the subversion we introduced the change in?

Reply:
It came in in 17.0, so the minor version wouldn't matter. We could make sure it isn't a preview before it was merged, but feels like more work than is warranted. Not a ton of people use previews, and if they're happy being on one preview, they can update to a later one.